Mathlib/LinearAlgebra/RootSystem/Basic.lean

```diff
@@ -74,6 +71,27 @@ protected def equiv_of_mapsTo :
 
 end reflection_perm
 
+lemma infinite_of_linearly_independent_coxeterWeight_four [CharZero R] [NoZeroSMulDivisors ℤ M]
+    (P : RootPairing ι R M N) (i j : ι) (hl : LinearIndependent R ![P.root i, P.root j])
+    (hc : P.coxeterWeight i j = 4) : Infinite ι := by
+  refine (infinite_range_iff (Embedding.injective P.root)).mp (Infinite.mono ?_
+    (infinite_range_of_reflection_reflection_iterate (P.coroot_root_two i)
```

Comment:
Some small change will be necessary here too which I will leave to you (assuming you agree to the suggestions above).